src/claviska/SimpleImage.php

```diff
@@ -15,2176 +16,2368 @@
+     * @param  int  $srcW Source width.
+     * @param  int  $srcH Source height.
+     * @return bool true if success.
+     */
+    protected static function imageCopyMergeAlpha($dstIm, $srcIm, int $dstX, int $dstY, int $srcX, int $srcY, int $srcW, int $srcH, int $pct): bool
+    {
+        // Are we merging with transparency?
+        if ($pct < 100) {
+            // Disable alpha blending and "colorize" the image using a transparent color
+            imagealphablending($srcIm, false);
+            imagefilter($srcIm, IMG_FILTER_COLORIZE, 0, 0, 0, round(127 * ((100 - $pct) / 100)));
+        }
+
+        imagecopy($dstIm, $srcIm, $dstX, $dstY, $srcX, $srcY, $srcW, $srcH);
+
+        return true;
+    }
+
+    /**
+     * Rotates an image so the orientation will be correct based on its exif data. It is safe to call
+     * this method on images that don't have exif data (no changes will be made).
+     *
+     * @return SimpleImage
+     *
+     * @throws Exception
+     */
+    public function autoOrient(): static
+    {
+        $exif = $this->getExif();
+
+        if (! $exif || ! isset($exif['Orientation'])) {
+            return $this;
+        }
+
+        switch($exif['Orientation']) {
+            case 1: // Do nothing!
+                break;
+            case 2: // Flip horizontally
+                $this->flip('x');
+                break;
+            case 3: // Rotate 180 degrees
+                $this->rotate(180);
+                break;
+            case 4: // Flip vertically
+                $this->flip('y');
+                break;
+            case 5: // Rotate 90 degrees clockwise and flip vertically
+                $this->flip('y')->rotate(90);
+                break;
+            case 6: // Rotate 90 clockwise
+                $this->rotate(90);
+                break;
+            case 7: // Rotate 90 clockwise and flip horizontally
+                $this->flip('x')->rotate(90);
+                break;
+            case 8: // Rotate 90 counterclockwise
+                $this->rotate(-90);
+                break;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Proportionally resize the image to fit inside a specific width and height.
+     *
+     * @param  int  $maxWidth The maximum width the image can be.
+     * @param  int  $maxHeight The maximum height the image can be.
+     * @return SimpleImage
+     */
+    public function bestFit(int $maxWidth, int $maxHeight): static
+    {
+        // If the image already fits, there's nothing to do
+        if ($this->getWidth() <= $maxWidth && $this->getHeight() <= $maxHeight) {
+            return $this;
+        }
+
+        // Calculate max width or height based on orientation
+        if ($this->getOrientation() === 'portrait') {
+            $height = $maxHeight;
+            $width = $maxHeight * $this->getAspectRatio();
+        } else {
+            $width = $maxWidth;
+            $height = $maxWidth / $this->getAspectRatio();
```

Comment:
Looks like we need to add `round()` to this to prevent the following error:

> PHP Deprecated:  Implicit conversion from float 375.234521575985 to int loses precision in /Users/claviska/SimpleImage/src/claviska/SimpleImage.php on line 958

I noticed this when using `bestFit()` in the example, as it was causing the image to not render. This might affect other methods in a similar way.

To replicate:

```sh
cd SimpleImage/example
php -S localhost:8080
```

Then a open a browser at http://localhost:8080/index.php and notice the broken image.